src/compiler/moduleNameResolver.ts

```diff
@@ -675,6 +675,13 @@ namespace ts {
     function tryLoadModuleUsingOptionalResolutionSettings(extensions: Extensions, moduleName: string, containingDirectory: string, loader: ResolutionKindSpecificLoader,
         state: ModuleResolutionState): Resolved | undefined {
 
+        const { baseUrl, paths } = state.compilerOptions;
+        if (baseUrl && paths && isEligibleForPathMapping(moduleName, paths)) {
+            if (state.traceEnabled) trace(state.host, Diagnostics.paths_option_is_specified_looking_for_a_pattern_to_match_module_name_0, moduleName);
+            const resolved = tryLoadModuleUsingPaths(extensions, moduleName, baseUrl, paths, loader, /*onlyRecordFailures*/ false, state);
+            if (resolved) return resolved.value;
+        }
+
         if (!isExternalModuleNameRelative(moduleName)) {

```

Comment:
Also needs some tests to ensure that caching is working as intended (and hasn't changed)

Reply:
Could you point me in the right direction? I am unsure what would be needed.

Reply:
There are tests in `src/testRunner/moduleResolution.ts` that verify caching is working as expected.

Reply:
@sheetalkamat I figured it out, they are run by default